tls/s2n_cbc.c

```diff
@@ -79,7 +72,8 @@ int s2n_verify_cbc(struct s2n_connection *conn, struct s2n_hmac_state *hmac, str
     int mismatches = s2n_constant_time_equals(decrypted->data + payload_length, check_digest, mac_digest_size) ^ 1;
 
     /* Compute a MAC on the rest of the data so that we perform the same number of hash operations */
-    POSIX_GUARD(s2n_hmac_update(copy, decrypted->data + payload_length + mac_digest_size, decrypted->size - payload_length - mac_digest_size - 1));
+    POSIX_GUARD(s2n_hmac_reset(hmac));
+    POSIX_GUARD(s2n_hmac_update(hmac, decrypted->data + payload_length + mac_digest_size, decrypted->size - payload_length - mac_digest_size - 1));
```

Comment:
This change makes sense: You only update and don't need to finalize, so resetting the state and always processing the remaining blocks serves to eliminate the timing signal while saving on memory.